tests/test_type/test_product.py

```diff
@@ -0,0 +1,10 @@
+import magma as m
+
+
+def test_anon_product_literal():
+    class Foo(m.Circuit):
+        io = m.IO(I=m.In(m.Bit), O=m.Out(m.Bit))
+
+        x = m.AnonProduct[{"I": m.Bit, "CE": m.Bit}]()
+        x @= m.product(I=io.I, CE=False)
+    assert Foo.x.CE.value() is m.GND
```

Comment:
nit
```suggestion
def test_anon_product_literal():

    class Foo(m.Circuit):
        io = m.IO(I=m.In(m.Bit), O=m.Out(m.Bit))
        x = m.AnonProduct[{"I": m.Bit, "CE": m.Bit}]()
        x @= m.product(I=io.I, CE=False)

    assert Foo.x.CE.value() is m.GND
```

actually do we need to have an enclosing circuit here? could we just do this with free values? e.g.

```python
a = m.Bit(name="a")
x = m.AnonProduct({"u": m.Bit, "v": m.Bit}](name="x")
x @= m.product(u=a, v=False)

assert x.v.value() is m.GND
```